### PR TITLE
All outgoing requests will now have a default User Agent

### PR DESCRIPTION
**Pull request by RishiDiwanTT**
## Description
Currently set to the minimalistic Palace Manager/1.x.x

## Motivation and Context
Overdrive have requested the CM have a custom User Agent while making requests to their endpoints.
[Notion](https://www.notion.so/lyrasis/Make-user-agent-less-generic-4820ea007b3c4e1988c569c3988f08c5)

## How Has This Been Tested?
Unit tests have been updated.
Used a mock server to inspect the user-agent manually.

## Checklist

- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/util/http.py [hunk 1]
```diff
@@ -252,16 +252,19 @@ def _request_with_timeout(
...
-                    k = k.encode("utf8")
-                if isinstance(v, str):
-                    v = v.encode("utf8")
-                new_headers[k] = v
-            kwargs["headers"] = new_headers
+
+        headers = kwargs.get("headers", {})
+        # Set a user-agent if not already present
+        if "User-Agent" not in headers:
+            headers["User-Agent"] = "Palace Manager/1.x.x"
```
Comment (dbernstein) [inline core/util/http.py:259]: How much work would it be to pull in the version (and potentially revision)?  Not sure if this is out of scope.

Comment (RishiDiwanTT) [reply core/util/http.py:259]: We currently don't have the concept of numbered versions (like semver), we only use a commit hash. Hence we are using this stopgap example for now.

Comment (jonathangreen) [reply core/util/http.py:259]: This should perhaps reference the version number that gets set in the configuration so it returns the same info as the heartbeat endpoint. 

https://github.com/ThePalaceProject/circulation/blob/dd910452b4e7883f580df1854a38977a9fc767ef/core/config.py#L522

Comment (tdilauro) [reply core/util/http.py:259]: @jonathangreen It was my suggestion to include a generic version string until we could discuss any security implications. But, if you're good with it, I have no objection.

Also, good to see you here!

Comment (RishiDiwanTT) [reply core/util/http.py]: Updated to use app_version.

**Review by tdilauro — COMMENTED** [review]
Requesting a small change, though it might require a change/addition to tests.

File: core/config.py [hunk 2]
```diff
@@ -185,6 +185,8 @@ class Configuration(ConfigurationConstants):
 
     EXCLUDED_AUDIO_DATA_SOURCES = "excluded_audio_data_sources"
 
+    # In case an app version is not present, we can use this version as a fallback
+    # for all outgoing http requests with a custom user-agent
```
Comment (tdilauro) [inline core/config.py:189]: Did you mean for all requests withOUT a custom ua here?

Comment (RishiDiwanTT) [reply core/config.py:189]: Meant "default custom" UA, but it didn't come out right. "Without" is clearer.

**Review by tdilauro — APPROVED** [review]
LGTM! 🎺
